tt_metal/hw/inc/dataflow_api.h

```diff
@@ -1311,9 +1311,10 @@ void noc_semaphore_set_multicast(
  * way of a synchronization mechanism. The same as *noc_async_write_multicast*
  * with preset size of 4 Bytes.
  *
- * With this API, you cannot send data only to the source node. That is, if
- * num_dests is 1 and the encoded destination nodes consist of the node
- * sending the data, then no data will be sent. The method call will be a no-op.
+ * Note: With this API, sending data only to the source node (when num_dests
+ * is 1) may result in unexpected behaviour. For some parameters, hangs have
+ * been observed. For other parameters, nothing happens. Consider using
```

Comment:
@bbradelTT do you have any example cases where nothing happens?
I was discussing with @TT-BrianLiu and he said when it doesn't hang, it always does the local copy.

Reply:
Yes, for noc_semaphore_set_multicast_loopback_src. That was the reason for the comment in the first place. See the edits for tt_eager/tt_dnn/op_library/bmm/kernels/dataflow/reader_bmm_tile_layout_in0_sender_receiver_padding_block_sharded.cpp https://github.com/tenstorrent/tt-metal/pull/9388 around line 200 (need to unhide) (code link: https://github.com/tenstorrent/tt-metal/blame/15fa272ccef0b19ab64926399f721c8feb4575dc/ttnn/cpp/ttnn/operations/matmul/device/kernels/dataflow/reader_bmm_tile_layout_in0_sender_receiver_padding_block_sharded.cpp#L210)

Reply:
Does `noc_semaphore_set_multicast_loopback_src` behave differently to `noc_async_write_multicast_loopback_src` for single core case? With linking set to False:
- `noc_async_write_multicast_loopback_src` does the local copy

Reply:
It shouldn't. The only differences between the two are the waypoint string and size vs 4.

Having said that, I don't have a good way to explore what happens in detail. Therefore I can only point to what we observed in the past. And the specific observation I have from before is for noc_semaphore_set_multicast_loopback_src

Reply:
Updated wording to be less definitive.